tests/einsum_test.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import autohoot.autodiff as ad
+import autohoot.backend as T
+from autohoot.graph_ops.graph_transformer import optimize
+from autohoot.utils import CharacterGetter
```

Comment:
I am a bit concerning exposing CharacterGetter in the tests. 

Can we start with some arbitrary character and then use it for loop directly?

Reply:
done